lib/spack/spack/schema/modules.py

```diff
@@ -224,6 +245,41 @@ def deprecation_msg_default_module_set(instance, props):
 }
 
 
+# deprecated keys and their replacements
+exclude_include_translations = {
+    "whitelist": "include",
+    "blacklist": "exclude",
+    "blacklist_implicits": "exclude_implicits",
+    "environment_blacklist": "exclude_environment",
+}
+
+
+def update_keys(data, key_translations, changed=False):
```

Comment:
This `changed` argument looks unnecessary.

Reply:
why?

Reply:
Because it is only used as a return value, not as an input. Everything that modifies it returns the value, and is `|=` into the final return. 

The sub-dict doesn't need to know about whether its parent changed to know what to return. If the parent changed, the parent flips the return code True, and if the child changed, it flips it.

Reply:
Ah ok -- true -- I'll get rid of the arg

Reply:
fixed